### PR TITLE
Ensure that `http-proxy-middleware` logs requests when log level is set to `debug`

### PR DESCRIPTION
**Pull request by dotboris**
## Hey, I just made a Pull Request!

This PR ensures that we pass the log level to `http-proxy-middleware`. Under the hood, `http-proxy-middleware` checks its log level to see if it should log requests. Passing in the log level ensures that we see those messages when we run with `LOG_LEVEL=debug`.

For reference, here's the log level checking behaviour from `http-proxy-middleware`: https://github.com/chimurai/http-proxy-middleware/blob/2.x/src/http-proxy-middleware.ts#L140

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/proxy-backend/src/service/router.ts [hunk 1]
```diff
@@ -121,6 +127,9 @@ export function buildMiddleware(
 
   // Attach the logger to the proxy config
   fullConfig.logProvider = () => logger;
+  if (isValidHttpProxyMiddlewareLogLevel(logger.level)) {
```
Comment (Rugvip) [inline plugins/proxy-backend/src/service/router.ts:130]: The ability to inspect the log level of the logger is about to disappear, so I'm hesitant to add this. Would it make sense to always pass a high log level here and then handle filtering or our end instead through the custom logger passed to the middleware?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you, makes sense! 👍 

Bit of a concern as we'll be switching over to the `LoggerService` from `@backstage/backend-plugin-api` in not too long, so wanna make sure we implement it in a way that is compatible with that.

File: plugins/proxy-backend/src/service/router.ts (hunk 1 above)
Comment (dotboris) [reply plugins/proxy-backend/src/service/router.ts]: Sounds good. I systematically pass in `debug` now (the most verbose option).

**Review by Rugvip — APPROVED** [review]
Thanks! 👍 

Let's :shipit:

**Pull request merged**